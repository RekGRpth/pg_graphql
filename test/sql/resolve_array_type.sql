begin;

    create table exotic(
        id bigint primary key,
        tags text[] not null
    );

    insert into exotic(id, tags)
    values (1, array['a', 'b']);

    select jsonb_pretty(
        graphql.resolve($$
            {
              exoticCollection {
                edges {
                  node {
                    id
                    tags
                  }
                }
              }
            }
        $$)
    );

rollback;
