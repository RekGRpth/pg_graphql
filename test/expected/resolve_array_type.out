begin;
    create table exotic(
        id bigint primary key,
        tags text[] not null
    );
    insert into exotic(id, tags)
    values (1, array['a', 'b']);
    select jsonb_pretty(
        graphql.resolve($$
            {
              exoticCollection {
                edges {
                  node {
                    id
                    tags
                  }
                }
              }
            }
        $$)
    );
            jsonb_pretty            
------------------------------------
 {                                 +
     "data": {                     +
         "exoticCollection": {     +
             "edges": [            +
                 {                 +
                     "node": {     +
                         "id": "1",+
                         "tags": [ +
                             "a",  +
                             "b"   +
                         ]         +
                     }             +
                 }                 +
             ]                     +
         }                         +
     },                            +
     "errors": [                   +
     ]                             +
 }
(1 row)

rollback;
